source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -382,45 +386,57 @@ BsonExtraElements Example
 
 The ``@BsonExtraElements`` annotation allows you to specify a field to
 deserialize data from a MongoDB document that lacks a corresponding POJO
-field mapping. Consider a situation in which your application needs to handle
-data that conforms to different data schema versions. When your POJO fields
-change to accommodate the schema, you can continue to read, but not
-update, the remainder of the MongoDB document data from the field that uses
-this annotation.
-
-Consider a situation in which you need to update the ``Product`` POJO from the
-:ref:`previous example <bson-annotation-code-example>` to accommodate the
-following changes to your data schema:
-
-- The ``name`` field is no longer needed, but must the data must remain for
-  legacy support.
-- The stock keeping unit (SKU) needs to be added.
-
-The updated version of the ``Product`` POJO should resemble the following code
-example:
+field mapping. This is useful when your application needs to work with data in
+a partially-defined schema. You can use this annotation to access data from
+any fields that do not correspond to the fields on your POJO.
+
+Consider a situation in which you store and retrieve data for an online store
+using the :ref:`Product POJO <bson-annotation-code-example>` from the previous
+example. As you offer a greater variety of products for the store, you
+discover that you need additional fields to describe them. Instead of
+mapping each additional field to the POJO, you can access them from a
+single field annotated with ``@BsonExtraElements`` as shown in the following
+code example:
 
 .. code-block:: java
-   :emphasize-lines: 6-7,9-10
+   :emphasize-lines: 12-13
 
    public class Product {
+       @BsonProperty("modelName")
+       private String name;
+
        @BsonId()
        @BsonRepresentation(BsonType.OBJECT_ID)
        private String serialNumber;
 
-       @BsonProperty("sku")
-       private String sku;
+       @BsonIgnore
+       private List<Product> relatedItems;
 
        @BsonExtraElements
-       private Document legacyInfo;
+       private Document additionalInfo;
 
        // ...
 
-Once you update the product records with SKU values in MongoDB, the data you
-retrieve with the ``Product`` POJO should resemble the following:
+If a database client updated the documents in the collection to include
+additional fields that describe the products such as ``dimensions`` and
+``weight``, a document retrieved using the ``Product`` POJO might resemble
+the following JSON object:
 
-.. code-block:: none
+.. code-block:: json
+   :copyable: false
+   :emphasize-lines: 5-9
+
+   {
+     "name": "MDB0123",
+     "serialNumber": "62e2...",
+     "relatedItems": null,
+     "additionalInfo": {
```

Comment:
This is not how it's supposed to work.  Rather, it should be:

```
   {
     "name": "MDB0123",
     "serialNumber": "62e2...",
     "relatedItems": null,
     "dimensions": "3x4x5",
     "weight": "256g"
   }
```

See [this test case](https://github.com/mongodb/mongo-java-driver/blob/95cb47a61f2896044e69ab73d59306e1527686e2/bson/src/test/unit/org/bson/codecs/pojo/PojoRoundTripTest.java#L528-L531) in the driver.

Do you have an example application that shows it working this way?

Reply:
Sorry, I think the disconnect is that I was portraying what was in the POJO, not MongoDB since the latter is redundant. I'll show the output using a generic toString() method to make it more apparent since that would show the Java class names.

Reply:
E.g.
```
ProductWithBsonExtraElements [name=MDB0123, serialNumber=62e2b9c5c32cb23e8cb0c95e, relatedItems=null, additionalInfo=Document{{dimensions=5x5x5}}]
```